core-web/libs/sdk/react/src/lib/components/DotEditableText/DotEditableText.tsx

```diff
@@ -0,0 +1,141 @@
+import { Editor } from '@tinymce/tinymce-react';
+import { useEffect, useRef, useState } from 'react';
+
+import {
+    isInsideEditor as isInsideEditorFn,
+    postMessageToEditor,
+    CUSTOMER_ACTIONS,
+    DotCmsClient
+} from '@dotcms/client';
+
+import { DotEditableTextProps, TINYMCE_CONFIG } from './utils';
+
+export function DotEditableText({
+    mode = 'plain',
+    format = 'text',
+    contentlet,
+    fieldName = ''
+}: DotEditableTextProps) {
+    const editorRef = useRef<Editor['editor'] | null>(null);
+    const [isInsideEditor, setisInsideEditor] = useState(false);
+    const [content, setContent] = useState(contentlet[fieldName] || '');
+
+    useEffect(() => {
+        setisInsideEditor(isInsideEditorFn());
+
+        if (!isInsideEditor) {
+            return;
+        }
```

Comment:
Do we need this? Maybe we can merge the 2 `useEffect`

Reply:
I separated them to make them more readable